x-pack/platform/plugins/shared/fleet/server/tasks/sync_integrations_task.ts

```diff
@@ -0,0 +1,221 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { SavedObjectsClient } from '@kbn/core/server';
+import type { CoreSetup, ElasticsearchClient, Logger } from '@kbn/core/server';
+import type {
+  ConcreteTaskInstance,
+  TaskManagerSetupContract,
+  TaskManagerStartContract,
+} from '@kbn/task-manager-plugin/server';
+import { getDeleteTaskRunResult } from '@kbn/task-manager-plugin/server/task';
+import type { LoggerFactory } from '@kbn/core/server';
+import { errors } from '@elastic/elasticsearch';
+
+import { SO_SEARCH_LIMIT, outputType } from '../../common/constants';
+import type { NewRemoteElasticsearchOutput } from '../../common/types';
+
+import { appContextService, outputService } from '../services';
+import { getInstalledPackageSavedObjects } from '../services/epm/packages/get';
+
+export const TYPE = 'fleet:sync-integrations-task';
+export const VERSION = '1.0.0';
+const TITLE = 'Fleet Sync Integrations Task';
+const SCOPE = ['fleet'];
+const INTERVAL = '5m';
+const TIMEOUT = '1m';
+export const FLEET_SYNCED_INTEGRATIONS_INDEX_NAME = 'fleet-synced-integrations';
+
+interface SyncIntegrationsTaskSetupContract {
+  core: CoreSetup;
+  taskManager: TaskManagerSetupContract;
+  logFactory: LoggerFactory;
+}
+
+interface SyncIntegrationsTaskStartContract {
+  taskManager: TaskManagerStartContract;
+}
+
+interface SyncIntegrationsData {
+  remote_es_hosts: Array<{
+    name: string;
+    hosts: string[];
+    sync_integrations: boolean;
+  }>;
+  integrations: Array<{
+    package_name: string;
+    package_version: string;
+    updated_at: string;
+  }>;
+}
+
+export class SyncIntegrationsTask {
+  private logger: Logger;
+  private wasStarted: boolean = false;
+  private abortController = new AbortController();
```

Comment:
Not sure I've seen an `AbortController` used like this - seems **quite** nice, I'll have to remember to point folks back to this!